### DISCUSSION TITLE
Personal Access Tokens expiration report

### DISCUSSION THREAD

**Discussion by johnmbaughman** [vinijmoura/Azure-DevOps#13] [General]
I recently found your Medium articles and GitHub repo and have been learning A LOT from this, so thank you!

I did have one issue with running the "tokenadmin/personalaccesstokens" API method. The code you have doesn't work when your DevOps instance is tied to an Azure Active Directory instance. 

To get that to work, you need to Base64 encode the username and the PAT together separated by a colon ":".  Example "johnb@company.com:MY_REGULAR_UNENCODED_PAT". 

How I did it was like this:

`$pat = "$(ConvertTo-Base64 -Value "$($Env:USERNAME)@company.com:$($env:AZURE_DEVOPS_EXT_PAT)" -Encoding  ([Text.Encoding]::ASCII))"`

Just thought I'd share that with you so that you could possibly update your example for future DevOps admins. 

Thank you again for all of these AWESOME examples! They are quite useful!

-John

**Comment by vinijmoura** [comment]
Thanks, @johnmbaughman for following me here!
Please, create a Pull Request that I merge your suggestion!

**Reply by johnmbaughman** [reply]
Sure, it'll take a little while, but I'll get one.
